src/Utils.ts

```diff
@@ -20,16 +20,24 @@ import * as path from 'path'
 export function getHost() {
     let config = workspace.getConfiguration('openhab')
     let host = config.host
-    let protocol = 'http'
     let port = config.port
+    let username = config.username
+    let password = config.password
 
+    let protocol = 'http'
+    let authentication = ''
+    
     if (host.includes('://')) {
         let split = host.split('://')
         host = split[1]
         protocol = split[0]
-    }
+    }    
 
-    return protocol + '://' + host + ':' + port
+    if (username.length > 0 && password.length > 0) {

```

Comment:
What if there's an username but no password?
Instead of the `if` you may write it like
```js
authentication = (username || '') + (password ? ':' + password : '')
authentication += authentication ? '@' : ''
```